### PR TITLE
bigquery: use client from Airflow hook when possible

### PR DESCRIPTION
**Pull request by mobuchowski**
This PR makes bigquery integration for Airflow reuse hook code that provides BigQuery client.

By doing that, we use "proven" connection instead of relying on `GOOGLE_APPLICATION_CREDENTIALS` which are one of the few different methods of specifying BigQuery connection in Airflow - for example, providing keyfile json in UI.

Additionally, this solves issue of using proper query location that @collado-mike raised.

Closes: https://github.com/OpenLineage/OpenLineage/issues/649

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

### CODE REVIEW

File: integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py [hunk 1]
```diff
@@ -64,7 +67,10 @@ def extract_on_complete(self, task_instance) -> Optional[TaskMetadata]:
                 }
             )
 
-        stats = BigQueryDatasetsProvider().get_facets(bigquery_job_id)
+        print(self.operator.__dict__)
```
Comment (collado-mike) [inline integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py:70]: Remove?

**Review by collado-mike — APPROVED** [review]
Nice

File: integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py (hunk 1 above)
Comment (mobuchowski) [reply integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py]: 😅

File: integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py [hunk 2]
```diff
@@ -80,6 +85,28 @@ def extract_on_complete(self, task_instance) -> Optional[TaskMetadata]:
             job_facets=job_facets
         )
 
+    def _get_client(self):
+        # Get client using Airflow hook - this way we use the same credentials as Airflow
+        if hasattr(self.operator, 'hook') and self.operator.hook:
+            hook = self.operator.hook
```
Comment (pankajkoti) [inline integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py:91]: I can see we're returning `Client()` on line 108. What do we do with this `hook`?

**Pull request merged**